pkg/models/datasource_cache.go

```diff
@@ -11,17 +11,87 @@ import (
 	"time"
 
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+var datasourceRequestCounter = prometheus.NewCounterVec(
```

Comment:
Just curious - are you hoping to measure success/failure rates with this counter (I'm assuming `code` refers to the response code)?

Reply:
Correct! :dart: